delphi-checks/src/main/java/au/com/integradev/delphi/checks/RedundantInheritedCheck.java

```diff
@@ -0,0 +1,122 @@
+import org.sonar.plugins.communitydelphi.api.check.DelphiCheck;
+import org.sonar.plugins.communitydelphi.api.check.DelphiCheckContext;
+import org.sonar.plugins.communitydelphi.api.symbol.declaration.RoutineKind;
+import org.sonar.plugins.communitydelphi.api.symbol.declaration.RoutineNameDeclaration;
+import org.sonar.plugins.communitydelphi.api.symbol.declaration.TypeNameDeclaration;
+import org.sonar.plugins.communitydelphi.api.type.Type;
+import org.sonar.plugins.communitydelphi.api.type.Type.ScopedType;
+
+@Rule(key = "RedundantInherited")
+public class RedundantInheritedCheck extends DelphiCheck {
+  private static final String MESSAGE = "Redundant inherited calls should be removed";
+
+  @Override
+  public DelphiCheckContext visit(RoutineImplementationNode routine, DelphiCheckContext context) {
+    List<DelphiNode> violationNodes = findViolations(routine);
+    for (DelphiNode violationNode : violationNodes) {
+      reportIssue(context, violationNode, MESSAGE);
+    }
+
+    return super.visit(routine, context);
+  }
+
+  private static List<DelphiNode> findViolations(RoutineImplementationNode routine) {
+    CompoundStatementNode block = routine.getStatementBlock();
+    if (block == null) {
+      return Collections.emptyList();
+    }
+
+    List<StatementNode> statements = block.getDescendentStatements();
+    if (statements.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    List<RoutineNameDeclaration> inheritedMethods = getParentMethodDeclarations(routine);
+    if (!inheritedMethods.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    return statements.stream()
+        .filter((statement) -> statement instanceof ExpressionStatementNode)
+        .map((statement) -> ((ExpressionStatementNode) statement).getExpression())
+        .filter(ExpressionNodeUtils::isBareInherited)
+        .collect(Collectors.toList());
+  }
+
+  private static Stream<Type> concreteParentTypesStream(Type type) {
+    return type.ancestorList().stream()
+        .filter(Type::isClass)
+        .findFirst()
+        .map(value -> Stream.concat(Stream.of(value), concreteParentTypesStream(value)))
+        .orElseGet(Stream::empty);
+  }
+
+  private static List<RoutineNameDeclaration> getParentMethodDeclarations(
+      RoutineImplementationNode method) {
+    TypeNameDeclaration typeDeclaration = method.getTypeDeclaration();
+    RoutineNameDeclaration nameDeclaration = method.getRoutineNameDeclaration();
+    if (typeDeclaration == null || nameDeclaration == null) {
+      return Collections.emptyList();
+    }
+
+    return concreteParentTypesStream(typeDeclaration.getType())
+        .map(ScopedType.class::cast)
+        .flatMap(type -> type.typeScope().getRoutineDeclarations().stream())
+        .filter(methodDeclaration -> isOverriddenMethod(methodDeclaration, nameDeclaration))
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  private static boolean isOverriddenMethod(
```

Comment:
This is suitably gnarly and complicated to get pulled out into a `delphi-checks` util, rather than copy-pasting.

Reply:
How do you feel about putting it in `au/com/integradev/delphi/utils/RoutineUtils.java`?

Reply:
`RoutineUtils` seems fine.